addons/zeus/functions/fnc_bi_moduleCurator.sqf

```diff
@@ -0,0 +1,328 @@
+                    default {
+                        _player = missionnamespace getvariable [_ownerVar,objnull];
+                    };
+                };
+
+                waituntil {unassigncurator _logic; isnull (getassignedcuratorunit _logic)};
+                waituntil {_player assignCurator _logic; getassignedcuratorunit _logic == _player};
+
+                //--- Add radio channels
+                {
+                    _x radiochanneladd [_player];
+                } foreach (_logic getvariable ["channels",[]]);
+
+                /* Removed by ACE
+                //--- Sent notification to all assigned players
+                {
+                    if (isplayer _x) then {
+                        [["CuratorAssign",[_name,name _player]],"bis_fnc_showNotification",_x] call bis_fnc_mp;
+                    };
+                } foreach (curatoreditableobjects _logic);
+                */
+
+                [_logic,"curatorUnitAssigned",[_logic,_player]] call bis_fnc_callscriptedeventhandler;
+
+                //--- Forced interface
+                //if (_forced) then {
+                //  [[true,true],"bis_fnc_forceCuratorInterface",_player] call bis_fnc_mp;
+                //};
+
+                //--- Wait for player to stop being Zeus
+                switch true do {
+                    case (_ownerUID > 0): {
+                        waituntil {
+                            sleep 0.01;
+                            {getplayeruid _x == _ownerVar} count playableunits == 0
+                        };
+                    };
+                    default {
+                        waituntil {_player != missionnamespace getvariable [_ownerVar,objnull]};
+                    };
+                };
+
+                //--- Add radio channels
+                {
+                    _x radiochannelremove [_player];
+                } foreach (_logic getvariable ["channels",[]]);
+
+                //--- Unassign
+                waituntil {unassigncurator _logic; isnull (getassignedcuratorunit _logic)};
+            };
+        };
+
+        /* Removed by ACE
+        //--- Create bird
+        _birdType = _logic getvariable ["birdType","eagle_f"];
+        if (_birdType != "") then {
+            _bird = createvehicle [_birdType,[100,100,100],[],0,"none"];
+            _logic setvariable ["bird",_bird,true];
+        };
+        */
+
+        //--- Activated all future addons
+        _addons = [];
+        {
+            if (typeof _x == "ModuleCuratorAddAddons_F") then {
+                _paramAddons = call compile ("[" + (_x getvariable ["addons",""]) + "]");
+                {
+                    if !(_x in _addons) then {_addons set [count _addons,_x];};
+                    {
+                        if !(_x in _addons) then {_addons set [count _addons,_x];};
+                    } foreach (unitaddons _x);
+                } foreach _paramAddons;
+            };
+        } foreach (synchronizedobjects _logic);
```

Comment:
This code only runs on the server, so JIP shouldn't be a problem (otherwise it'd be a wonder that zeus ever worked).